### PR TITLE
Bring Python client up-to-date

### PR DESCRIPTION
**Pull request by ElliotGregory**
These changes bring the mqlight python client upto the same level as the other clients.
There have been minor changes to the external API whereby some callback functions now include the client object.  Other changes include TSL support and extended parameter checking and unit test support.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -1,13 +1,13 @@
 # <copyright
 # notice="lm-source-program"
 # pids="5725-P60"
-# years="2013,2014"
+# years="2013,2016"
```
Comment (mimaison) [inline docs/conf.py:4]: Should the date be updated to 2017 ?

File: docs/sample.rst [hunk 2]
```diff
@@ -35,38 +33,58 @@ optional arguments:
                         seconds (default: None)
   -n SHARE_NAME, --share-name SHARE_NAME
                         optionally, subscribe to a shared destination using
-                        SHARE_NAME as the share name.
+                        SHARE_NAMEas the share name.
```
Comment (mimaison) [inline docs/sample.rst:36]: missing space

File: docs/sample.rst [hunk 3]
```diff
@@ -35,38 +33,58 @@ optional arguments:
...
+ssl arguments:
+  -c FILE, --trust-certificate FILE
+                        use the certificate contained in FILE (in PEM or DER
+                        format) to validate the identify of the server. The
+                        connection must be secured with SSL/TLS (e.g. the
+                        service URL must start with "amqps://")
+  --client-certificate  FILE
+                        use the certificate contained in FILE (in PEM format)
+                        to supply the identity of the client. The connection
+                        mustbe secured with SSL/TLS
```
Comment (mimaison) [inline docs/sample.rst]: "mustbe" -> "must be"

File: docs/sample.rst [hunk 4]
```diff
@@ -35,38 +33,58 @@ optional arguments:
...
+               [--client-key FILE] [--client-key-passphrase PASSPHRASE]
+               [--no-verify-name SSL_VERIFY_NAME]
+               [MESSAGE [MESSAGE ...]]
+
+
+Send a message to a MQ Light server.
 
 positional arguments:
-  MESSAGE               message to be sent (default: ['Hello world !'])
+  MESSAGE               message to be sent (default: None)
```
Comment (mimaison) [inline docs/sample.rst]: As far as I can tell the send.py sample has not been changed and still sends "Hello world!" by default

File: tests/test_createclient.py [hunk 5]
```diff
@@ -337,14 +337,14 @@ def test_invalid_ssl_options(self):
...
-                'ssl_verify_name': False
-            },
+#            {
+#                'ssl_trust_certificate': 'BadCertificate',
+#                'ssl_verify_name': True
+#            },
+#            {
+#                'ssl_trust_certificate': 'BadCertificate',
+#                'ssl_verify_name': False
+#            },
```
Comment (mimaison) [inline tests/test_createclient.py]: Is this commented out on purpose ?

**Review by mimaison — CHANGES_REQUESTED** [review]
Mostly nitpicks, thanks Elliot

File: docs/sample.rst (hunk 2 above)
Comment (ElliotGregory) [reply docs/sample.rst:36]: Has been corrected, thanks.

File: docs/conf.py (hunk 1 above)
Comment (ElliotGregory) [reply docs/conf.py:4]: I am leaving as 2016 as that was when the changes were made and built.

File: docs/sample.rst (hunk 3 above)
Comment (ElliotGregory) [reply docs/sample.rst]: Has been corrected, thanks.

File: docs/sample.rst (hunk 4 above)
Comment (ElliotGregory) [reply docs/sample.rst]: Has been corrected, thanks.

File: tests/test_createclient.py (hunk 5 above)
Comment (ElliotGregory) [reply tests/test_createclient.py]: Comments remove and retested - no issues detected, thanks.

**Review by mimaison — APPROVED** [review]

**Review by dnwe — APPROVED** [review]

**Comment by dnwe** [comment]
Thanks! Merged as 6137d85a59718de6dd4d16d6de254002345fe17a

**Pull request closed without merging**